Add table for caching failed process_manager.trigger requests

nh758 opened the pull request:
See:
https://github.com/digi-serve/ab_service_appbuilder/pull/18

Hiro-Nakamura commented:
The real question is now that we have the migrations scripts, do we continue to update the initial `.sql` implementation?  

Or is it better for the initial sql to be an expected base, that our migrations can run on, and we simply perform a migration run on a fresh install to then bring everything up to date?  

When I implemented the `switcheroo` changes, I didn't update the base sql.  I had intended for the migrations to be a simpler method of adding these changes rather than going back to the initial sql and trying to keep adding the new changes.

Also, if we are going to be trying to keep the `sql` files up to date, then we also have to change the sql in the `tenant_manager`/`utils`/`new_tenant.sql` as well.  But that one is tricky since it has placeholders in the sql data for the new `admin` info.  So we can't just do a simple SQL dump of an up to date site and paste in the new values.

Unless you can see a problem with this approach, then my vote would be to leave the `.sql` files as an expected base value and then run the migration on any new site / tenant we create.

Hiro-Nakamura commented:
if we do decide to keep the sql files up to date, then I suggest we also include the `SITE_CONFIG` table for the migrations in the sql files, and then keep the initial entry up to date with the latest patch so we don't have to re-run those patches for a site that has been initialized with the sql files.

nh758 commented:
> The real question is now that we have the migrations scripts, do we continue to update the initial `.sql` implementation?
> 
> Or is it better for the initial sql to be an expected base, that our migrations can run on, and we simply perform a migration run on a fresh install to then bring everything up to date?
> 
> When I implemented the `switcheroo` changes, I didn't update the base sql. I had intended for the migrations to be a simpler method of adding these changes rather than going back to the initial sql and trying to keep adding the new changes.
> 
> Also, if we are going to be trying to keep the `sql` files up to date, then we also have to change the sql in the `tenant_manager`/`utils`/`new_tenant.sql` as well. But that one is tricky since it has placeholders in the sql data for the new `admin` info. So we can't just do a simple SQL dump of an up to date site and paste in the new values.
> 
> Unless you can see a problem with this approach, then my vote would be to leave the `.sql` files as an expected base value and then run the migration on any new site / tenant we create.

Yes I had been questioning this myself. The main reason I added here is that we aren't running the migrations on new installs. The concern is mostly for the tests, if they expect tables to be current. 

I agree that these sql file are hard to maintain, so I'm in favor of adding the migrations as an install step, or into to ci test setups.
